model/dependency.go

```diff
@@ -91,6 +91,30 @@ func (di *dependencyIncluder) handle(pair TVPair, activationInfo *specificActiva
 		return true, nil
 	}
 
+	// For a task group task inside a single host task group, the previous
+	// tasks in the task group are implicit dependencies of the current task.
+	if tg := di.Project.FindTaskGroupForTask(pair.Variant, pair.TaskName); tg != nil && tg.MaxHosts == 1 {
+		catcher := grip.NewBasicCatcher()
+		for _, t := range tg.Tasks {
+			// When we reach the current task, stop recursing.
+			if t == pair.TaskName {
+				break
+			}
+			_, err := di.handle(TVPair{pair.Variant, t}, activationInfo, generatedVariants, false)
+			catcher.Wrapf(err, "task group '%s' in variant '%s' contains unschedulable task '%s'", pair.TaskName, pair.Variant, t)
+		}
+
+		// If any of the previous tasks in the task group are unschedulable, return false.
+		// And unschedule everything.
```

Comment:
"unschedule everything" -> "unschedule the earlier single host task group tasks"?

Reply:
I originally made it unschedule everything before the task that failed but I don't think the expected behavior should still be run some of the task group (if it failed in scheduling any earlier one, it should unschedule everything). I ended up removing it since it complicated up the logic a lot and since an error propogates, it won't schedule.

I'll specify "unschedule everything in the single host task group"